test/tests.bats

```diff
@@ -245,7 +245,7 @@ teardown_file() {
 }
 
 @test "checking smtp: delivers mail to existing account" {
-  run docker exec mail /bin/sh -c "grep 'postfix/lmtp' /var/log/mail/mail.log | grep 'status=sent' | grep ' Saved)' | sed 's/.* to=</</g' | sed 's/, relay.*//g' | sort | uniq -c | tr -s \" \""
+  run docker exec -e SPAMASSASSIN_SPAM_TO_INBOX=0 mail /bin/sh -c "grep 'postfix/lmtp' /var/log/mail/mail.log | grep 'status=sent' | grep ' Saved)' | sed 's/.* to=</</g' | sed 's/, relay.*//g' | sort | uniq -c | tr -s \" \""
```

Comment:
You made two changes here to call the `docker exec` commands with a different ENV value. This won't work as we're only using `grep` to search logs, and this ENV is for the startup scripts, not changeable at runtime.

Revert this change and update the tests expected outputs as detailed in the main review feedback.

---

<details>
<summary>Previous review comment ,regarding ENV testing and migration out of tests.bats file</summary>

These two tests could be extracted into the spam test I rewrote (`mail_spam_bounced.bats`), since that makes it easier to run a container for different tests with slight changes.

As these both require the value to be `0` (do they?), you could reference the remaining test in `mail_spam_bounced.bats`. The rejection test case seems to be the same, except slightly better in that it filters for a specific address, which AFAIK is searching content from [`amavis-spam.txt` template](https://github.com/docker-mailserver/docker-mailserver/blob/master/test/test-files/email-templates/amavis-spam.txt), the existing test in `mail_spam_bounced.bats` sends this already, so would just involve updating the current `grep` there to pipe one more to match `external.tld=spam@my-domain.com`, and perhaps the other pipe for `wc -l` with `assert_output 1` to verify it's just one matching line found.

This `delivers mail to existing account` test I'm not familiar with and probably would be it's own separate test case. It's less specific on what it's looking for, probably requires sending another mail, then it would be similar to `_should_bounce_spam()` test method, adjusting the template `nc` sends, and then the `docker logs` search query to what we have here in `tests.bats`. (_**EDIT:** Seems a little bit tricky, since that test looks like it's covering multiple mails sent earlier, I haven't verified if that's from earlier tests, or all at initial test setup_)

</details>